app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesViewModel.kt

```diff
@@ -29,33 +39,23 @@ import kotlinx.coroutines.*
 @FlowPreview
 class SearchRepositoriesViewModel(private val repository: GithubRepository) : ViewModel() {
 
-    companion object {
-        private const val VISIBLE_THRESHOLD = 5
-    }
-
-    private val queryLiveData = MutableLiveData<String>()
-    val repoResult: LiveData<RepoSearchResult> =  queryLiveData.switchMap {
-        liveData {
-            val repos = repository.getSearchResultStream(it).asLiveData(Dispatchers.Main)
-            emitSource(repos)
-        }
-    }
+    @Volatile
+    private var lastQueryValue: String? = null
+    @Volatile
+    private var lastSearchResult: Flow<PagingData<Repo>>? = null
 
     /**
      * Search a repository based on a query string.
      */
-    fun searchRepo(queryString: String) {
-        queryLiveData.postValue(queryString)
-    }
-
-    fun listScrolled(visibleItemCount: Int, lastVisibleItemPosition: Int, totalItemCount: Int) {
-        if (visibleItemCount + lastVisibleItemPosition + VISIBLE_THRESHOLD >= totalItemCount) {
-            val immutableQuery = queryLiveData.value
-            if (immutableQuery != null) {
-                viewModelScope.launch {
-                    repository.requestMore(immutableQuery)
-                }
-            }
+    fun searchRepo(queryString: String): Flow<PagingData<Repo>> {
+        val result = lastSearchResult
```

Comment:
consider renames:

`lastSearchResult` -> `currentSearchResult` (because it's still current - it's only the 'last' one in the scope of searchRepo)
`lastQueryValue` -> `currentSearchQuery` (clearer pairing with result)
`result` -> `lastResult` (since it's not the result much of the time)